Fix resolving devices by defining the correct search key

Talkabout opened the pull request:
When defining a device, for example, in the address module, it fails due to the wrong search key when calling `find_by_key` in `_resolve_entity`. 

Due to the fields needed for a device entity in phpIPAM API:

```
$ curl -sk -XGET -H "phpipam-token: ${PHPIPAM_TOKEN}" ${PHPIPAM_API_URL}/tools/devices/ | jq '.data[0] | {id: .id, hostname: .hostname, ip: .ip}'
{
  "id": "8",
  "hostname": "Test device",
  "ip": "192.2.0.222"
}
```

By introducing an extra case for controller `tools/devices` we can override the default search key with `hostname`.

Talkabout committed:
fixed resolving devices by defining the correct search key

cmeissner committed:
Add test and changelog entry

To guarantee that the introduced change work as expected, a test case
`address_device_mapping` was added too.
Second a short fragment for the changelog was added.

cmeissner commented:
@Talkabout, thank you for filing this pull request. I updated the description a bit for others to better follow what it will achieve. 

I also added a test case for that special combination of address and device to be sure the code is working. Also, a short changelog fragment was added to have a corresponding entry in the changelog of the upcoming version.

After granting you the permissions to trigger actions to run on push, I also realized that something seems to be broken. I will try to investigate on this issue, if I found time. You are also invited to invest some time to get to the root cause of that failure in CI.

cmeissner committed:
Increase the amount of test runners

Due to timing problems with running tests in parallel with only 4
runners we increases the amount of runners to 8 to mitigate conflicting
actions.
This is only a workaround and the tests need to be refactored to prevent
conflicting actions by using random data.

Talkabout commented:
@cmeissner Thanks for taking care!
